news/_posts/2019-09-11-prestashop-in-2019-and-beyond-part-2-pain-points.md

```diff
@@ -0,0 +1,471 @@
+	
+**The problem:** Not only do we have two classes that do the same thing, but they do it in a _different_ way and produce _different_ side effects. Which one should we use, and when?
+
+**3. Little or no added value**
+
+Looking at the code, we can see that many hastily made Adapters have been designed [as dumb bridges to legacy code][bad-example], even replicating method signatures and design flaws of their legacy counterparts. Some even have static methods!
+
+**The problem:** Many adapters do not even add any value to legacy classes and exist only to comply with the "no use of legacy classes in new code".
+
+**4. Hard dependencies**
+
+To make matters worse, most Adapters do not even implement an interface, which makes it impossible to replace them with Core reimplementations without modifying the dependent classes.
+
+**The problem:** Most adapters are hard dependencies and cannot be replaced without changing the dependent code. 
+
+**Bonus: No single source of truth**
+
+And finally, there's the matter of Database access. Since most interactions with the database are performed through ObjectModels, and since ObjectModels are too difficult to reimplement with an Adapter (we want to get rid of them anyway), some developers reimplemented them using Doctrine.
+
+**The problem:** Even models have been duplicated. Some of them _aren't even in the adapter namespace_ and work in a _completely different way_... so there's no single source of truth anymore.
+
+There are many other duplicate systems in 1.7:
+
+- Symfony controllers vs Legacy controllers
+- Twig vs Smarty
+- Symfony translation system vs legacy
+- New BO theme (based on bootstrap 4) for Symfony pages vs legacy one (based on bootstrap 3) for legacy pages
+
+### Legacy layout
+
+Contrary to what it might seem, Symfony-based BO pages are not fully independent from the legacy stack. In fact, not at all: whenever a Symfony controller is loaded, the whole legacy stack (including Smarty) is loaded as well. 
+
+Why? Because the main page layout (top menu & sidebar) hasn't been migrated yet and is handled by the legacy stack. Therefore all the content generated by the symfony controller is actually being rendered by a legacy controller called [LegacyLayout][legacy-layout].
+
+Having it done this way instead of the other way around (wrapping legacy controllers in Symfony) means that as the migration progresses, complexity increases.
+
+### Translation system
+
+PrestaShop 1.7 introduced the Symfony translator, which is used for all Core and Native modules (the ones bundled with PrestaShop). While the translator itself is better than the system it replaced, its implementation has some side effects:
+
+1. The translation catalogue is bundled with PrestaShop and frozen at the time of release, meaning that any new wordings cannot be added until the following release.
```

Comment:
I think we need to clarify "any new wordings" meaning, to make sure people dont think "custom translations is totally undoable with Symfony"

Reply:
Done